sdk/provisioning/Azure.Provisioning.CloudMachine/src/CloudMachineInfrastructure.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Provisioning.Authorization;
+using Azure.Provisioning.EventGrid;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Resources;
+using Azure.Provisioning.Roles;
+using Azure.Provisioning.ServiceBus;
+using Azure.Provisioning.Storage;
+
+namespace Azure.Provisioning.CloudMachine;
+
+public class CloudMachineInfrastructure : Infrastructure
+{
+    private readonly string _name;
+    private UserAssignedIdentity _identity;
+    private StorageAccount _storage;
+    private BlobService _blobs;
+    private BlobContainer _container;
+    private ServiceBusNamespace _serviceBusNamespace;
+    private ServiceBusNamespaceAuthorizationRule _serviceBusNamespaceAuthorizationRule;
+    private ServiceBusTopic _serviceBusTopic_main;
+    private ServiceBusTopic _serviceBusTopic_app;
+    private ServiceBusSubscription _serviceBusSubscription_main;
+    private ServiceBusSubscription _serviceBusSubscription_app;
+    private SystemTopic _eventGridTopic_Blobs;
+    private SystemTopicEventSubscription _systemTopicEventSubscription;
+
+    /// <summary>
+    /// The common principalId parameter.
+    /// </summary>
+    public BicepParameter PrincipalIdParameter => new BicepParameter("principalId", typeof(string));
+
+    /// <summary>
+    /// The common principalType parameter.
+    /// </summary>
+    public BicepParameter PrincipalTypeParameter => new BicepParameter("principalType", typeof(string));
+
+    /// <summary>
+    /// The common principalName parameter.
+    /// </summary>
+    public BicepParameter PrincipalNameParameter => new BicepParameter("principalName", typeof(string));
+
+    public CloudMachineInfrastructure(string? name = "cm") : base(name!)
+    {
+        _name = name ?? "cm";
```

Comment:
It feels a little weird to make it nullable but give it a default but then add an explicit override here.  I think this would be cleaner with a non-nullable param that defaults to `"cm"`.